cql3/expr/expression.cc

```diff
@@ -952,6 +952,61 @@ std::ostream& operator<<(std::ostream& s, oper_t op) {
     __builtin_unreachable();
 }
 
+std::vector<expression> extract_column_restrictions(const expression& expr, const column_definition& column) {
+    struct visitor {
+        std::vector<expression> restrictions;
+        const column_definition& column;
+        const binary_operator* current_binary_operator;
+
+        void operator()(bool b) {
+            restrictions.emplace_back(b);
+        }
+
+        void operator()(const conjunction& conj) {
+            for (const expression& child : conj.children) {
```

Comment:
I notice this in a lot of visitors.  I wonder if there's a good way to DRY it.  Unsurprisingly, I prefer this shorter version:
https://github.com/scylladb/scylla/blob/aaf35b5ac273afd01a73c93c09f27179bdea17b5/cql3/restrictions/statement_restrictions.cc#L241

Reply:
`class my_visitor : public recursive_visitor<void> { ... /* no need to implement operator(const conjunction&) -> void */ } ;

Reply:
I feel that this is gonna become so unreadable. Maybe we could just keep it stupid simple?
And the `for_each` version is actually longer than a simple for:
```c++
for (const expression& child : conj.children) {
    std::visit(*this, child);
}

std::ranges::for_each(conj.children, [this] (const expression& child) {
    std::visit(*this, child);
}
```